src/classes/CRLP_RollupProcessor_TEST.cls

```diff
@@ -0,0 +1,314 @@
+        System.assertEquals(false, processor.isOppAlreadyProcessedForParent(accountId, oppId2));
+        System.assertEquals(1, processor.getPreviouslyProcessedOppIds(accountId).size());
+    }
+
+    /**
+    * @description Validate the Parameter Validation method throws exceptions for missing data
+    */
+    @IsTest
+    private static void shouldHandleRequiredParameterValidation() {
+        CMT_UnitTestData_TEST.mockFullSetOfRollupDefinitions();
+
+        Account account = new Account(
+                Id = UTIL_UnitTestData_TEST.mockId(Account.SObjectType),
+                Name = 'TestAccount'
+        );
+
+        CRLP_RollupProcessor processor = new CRLP_RollupProcessor()
+                .withBatchJobMode(CRLP_RollupProcessingOptions.BatchJobMode.NonSkewMode)
+                .withSummaryRecords(new List<Account>{ account });
+
+        // Test with a missing Rollup Type
+        try {
+            processor.validateProcessSettingsBeforeStarting();
+            System.assert(false, 'An exception should not have been thrown');
+        } catch (Exception ex) {
+            System.assert(ex.getMessage().containsIgnoreCase('RollupType'),
+                    'A CRLP_RollupException exception should have been thrown: ' + ex.getMessage());
+        }
+
+        // Test with a missing Parent (Summary) record
+        processor.withSummaryRecords(null);
+        processor.withRollupType(CRLP_RollupProcessingOptions.RollupType.AccountHardCredit);
+        try {
+            processor.validateProcessSettingsBeforeStarting();
+            System.assert(false, 'An exception should not have been thrown');
+        } catch (Exception ex) {
+            System.assert(ex.getMessage().containsIgnoreCase('Summary'),
+                    'A CRLP_RollupException exception should have been thrown: ' + ex.getMessage());
+        }
+
+        // Test with both required parameters provided.
+        processor.withSummaryRecords(new List<Account>{ account });
+        processor.validateProcessSettingsBeforeStarting();
+        System.assert(true, 'An exception should not have been thrown');
+    }
+
+    /**
+    * @description Validate the cleanDetailOppsListForSoftCreditMode() method removes the expected
+    * previously processed detail records
+    */
+    @IsTest
+    private static void shouldRemoveSummaryRecordWithNoDetailRecords() {
+
+        List<Contact> contacts = new List<Contact>{
+            new Contact(Id = UTIL_UnitTestData_TEST.mockId(Contact.SObjectType)),
+            new Contact(Id = UTIL_UnitTestData_TEST.mockId(Contact.SObjectType)),
+            new Contact(Id = UTIL_UnitTestData_TEST.mockId(Contact.SObjectType))
+        };
+        List<Id> oppIds = UTIL_UnitTestData_TEST.mockIds(Opportunity.SObjectType, 3);
+
+        // Create same set of 3 PSC records for each Contact and Opportunity
+        List<Partial_Soft_Credit__c> detailRecords = new List<Partial_Soft_Credit__c>();
+        for (Integer ncon =0; ncon<3; ncon++) {
```

Comment:
Difficult to differentiate between `ncon` and `nopp`.

Reply:
Agreed. I was just noticing that too. I renamed to `numOpp` and `numContact`.